### PR TITLE
Fix mypy type checking errors

### PR DESCRIPTION
**Pull request by attemoi**
## Description

<https://jira.lingsoft.fi/browse/SIMPLYE-202> - Enable Mypy type checks for ekirjasto-circulation in Github

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/authenticator.py [hunk 1]
```diff
@@ -593,12 +594,14 @@ def create_bearer_token(
             # Maybe we should use something custom instead.
             iss=provider_name,
         )
-        return jwt.encode(payload, self.bearer_token_signing_secret, algorithm="HS256")
+        return jwt.encode(
+            payload, cast(str, self.bearer_token_signing_secret), algorithm="HS256"
```
Comment (attemoi) [inline api/authenticator.py:598]: This cast has been introduced in upstream as well.

File: api/ekirjasto_authentication.py [hunk 2]
```diff
@@ -469,10 +476,6 @@ def remote_refresh_token(self, token: str) -> (str, int):
             # Do nothing if authentication fails, e.g. token expired.
             return INVALID_EKIRJASTO_TOKEN, None
         elif response.status_code != 200:
-            msg = "Got unexpected response code %d. Content: %s" % (
```
Comment (attemoi) [inline api/ekirjasto_authentication.py:472]: this `msg` is not referenced anywhere

File: tests/finland/test_ekirjasto.py [hunk 3]
```diff
@@ -311,19 +311,12 @@ def test_secrets(
     ):
         provider = create_provider()
 
-        # Secrets are not set, so this will fail.
```
Comment (attemoi) [inline tests/finland/test_ekirjasto.py:314]: The case of secrets not being set are also covered by the test cases following this one, so removed from here to fix mypy errors.

**Review by ttuovinen — APPROVED** [review]
Looking good 👍

**Pull request merged**